hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java

```diff
@@ -508,6 +520,71 @@ private boolean isZombie(K id) {
 		return prop -> prop.getter().apply(defaultEntity);
 	}
 
+	private List<AccountChanges<K, A, P>> getCurrentAccountChanges() {
+		final List<AccountChanges<K, A, P>> accountChanges = new ArrayList<>();
+		for (final var changesEntry : changes.entrySet()) {
+			final var accountId = changesEntry.getKey();
+			final var accountProperties = changesEntry.getValue();
+			final var changesForSingleAccount = new AccountChanges<>(accountId, toGetterTarget(accountId),
```

Comment:
Ideally the `A` instances in this `accountChanges` list will be mutable---that is, derive from the expression,
```
entities.contains(id) ? entities.getRef(id) : newEntity.get()
```
...since otherwise we must acquire locks twice, once getting these immutable references; and again to get the mutable references in `getFinalized()`.